### PR TITLE
(DOCSP-28887): Kotlin IA Revamp - Realm Database section

### PR DESCRIPTION
**Pull request by cbullinger**
## Pull Request Info

Update existing Realm Database section: 
- Make Realm Database container into a landing page
- Remove Overview page and merge content to landing page 
- Move and rename `Open and Close a Realm` > `Configure & Open a Realm` 
- Make Manage Realm Files container into a landing page with overview content
- Create new Database Internals page

### Jira

- https://jira.mongodb.org/browse/DOCSP-28887

### Staged Changes

- [landing page] [Realm Database](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-28887-ia-realm-database/sdk/kotlin/realm-database/)
- [landing page] [Manage Realm Files](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-28887-ia-realm-database/sdk/kotlin/realm-database/realm-files/)
- [new] [Database Internals](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-28887-ia-realm-database/sdk/kotlin/realm-database/database-internals/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/kotlin/realm-database.txt [hunk 1]
```diff
@@ -1,26 +1,111 @@
...
-   :class: singlecol
-
 .. toctree::
    :titlesonly:
    
-   Overview </sdk/kotlin/realm-database/overview>
    Frozen Architecture </sdk/kotlin/realm-database/frozen-arch> 
    Model Data </sdk/kotlin/realm-database/schemas>
-   Manage Realm Database Files </sdk/kotlin/realm-database/realm-files>
+   Configure & Open a Realm </sdk/kotlin/realm-database/realm-files/open-and-close-a-realm>
```
Comment (dacharyc) [inline source/sdk/kotlin/realm-database.txt:13]: Curious why we're moving the "Configure & Open a Realm" page _out_ of "Manage Realm Files" as configuring & opening a realm seems to be the fundamental component of managing a realm file?

File: source/sdk/kotlin/realm-database.txt [hunk 2]
```diff
@@ -1,26 +1,110 @@
...
    Model Data </sdk/kotlin/realm-database/schemas>
-   Manage Realm Database Files </sdk/kotlin/realm-database/realm-files>
    Write Transactions </sdk/kotlin/realm-database/write-transactions>
    Create </sdk/kotlin/realm-database/create>
    Read </sdk/kotlin/realm-database/read>
    Update </sdk/kotlin/realm-database/update>
    Delete </sdk/kotlin/realm-database/delete>
    React to Changes </sdk/kotlin/realm-database/react-to-changes>
    Serialization </sdk/kotlin/realm-database/serialization>
+   Manage Realm Files </sdk/kotlin/realm-database/realm-files>
```
Comment (dacharyc) [inline source/sdk/kotlin/realm-database.txt:20]: One thing - I think if we put Configure & Open Realms in the "Manage Realm Files" section, we probably shouldn't move it to the bottom of the ToC. Maybe move this up?

**Review by dacharyc — APPROVED** [review]
Overall LGTM. Just one small Q about the Manage Realm Files section.

File: source/sdk/kotlin/realm-database.txt (hunk 2 above)
Comment (cbullinger) [reply source/sdk/kotlin/realm-database.txt]: Yep! Moved it up

**Pull request merged**